format_spec.md

```diff
@@ -0,0 +1,399 @@
+# MODIP Format Specification
+
+This document provides detailed information on the MODIP Format.
+
+The MODIP format uses JSON to store information. If providing an API that returns format-compliant information, the `application/json` Content-Type header SHOULD be provided. If stored on disk, the `.modip.json` extension SHOULD be used.
+
+A format-compliant example is provided in **examples/format_example.modip.json**.
+
+## Fields
+
+Fields are stored inside the JSON metadata of a Project. Fields that are optional will be marked as such. All fields are stored as a String value unless stated otherwise. Below is a complete list of all fields available for the MODIP Format, grouped into categories.
+
+## General Project Information
+
+### `schemaVersion`
+
+This field MUST be stored as an String. It contains the version of the Format being used, to help with backwards compatibility. The format is ALWAYS stored as a SemVer-conforming String. The current value of this at the time of writing is `1.0.0`.
+
+---
+
+### `id`
+
+The ID of the Project. There is no standard or required format on how to store an ID, other than it CANNOT include spaces. It may be a slug, UUID, random number, or other method of storing IDs. It is RECOMMENDED that IDs are not tied to the name of a project.
```

Comment:
Everything else is fine? How about U+00A0 NO-BREAK SPACE? U+2003 EM SPACE? U+1680 OGHAM SPACE MARK (this is a space -->  )? Newlines? Null characters? BELs?

Reply:
Yeah... I didn't put much thought into this. Perhaps it should be limited to something like `a-zA-Z0-9_-`. It may not be ideal, especially for non-Latin languages, though.